### ISSUE TITLE
[bug] Do not use old verifier versions inside the generator workflows

### ISSUE THREAD

**Issue by asraa** [slsa-framework/slsa-github-generator#1175]
**Describe the bug**
Using old versions of the verifier inside the `generate-builder` step with `compile-builder: false` results in hard-to-manage issues with verifier updates and incompatibilities.

Is there any way we can avoid this issue, either by 
* building the verifier from source (it's not as heavy-weight as the generator)?
* somehow securely detecting latest patch versions or latest releases?

Basically, embedding the verifier version inside the workflows has caused a lot of update process issues, like when slsa-verifier required an update to handle changes in production services.
https://github.com/slsa-framework/slsa-github-generator/issues/1163

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by ianlewis** [comment]
Moved milestones as, while not directly sigstore related, still helps with the stability issues w/ sigstore and the verifier.

**Comment by ianlewis** [comment]
> **Describe the bug** Using old versions of the verifier inside the `generate-builder` step with `compile-builder: false` results in hard-to-manage issues with verifier updates and incompatibilities.
> 
> Is there any way we can avoid this issue, either by
> * building the verifier from source (it's not as heavy-weight as the generator)?

I'm not sure this solves the issue unless we do something like build from `main`?

> * somehow securely detecting latest patch versions or latest releases?

In an ideal world we'd just use `slsa-verifier`'s [installer action](https://github.com/slsa-framework/slsa-verifier/tree/main/actions/installer). Unfortunately the installer downloads the verifier version referenced by the action's tag reference and doesn't download the latest version.

I know we wanted to keep the action code and verifier in sync with each other but I think we could revisit how that action is supposed to work. Maybe add an `install-latest: true` type of flag input? That should work since we bootstrap and verify the verifier's provenance on each release. As long as that process doesn't change it should be ok...

@laurentsimon @kpk47 wdut?

**Comment by asraa** [comment]
> I'm not sure this solves the issue unless we do something like build from main?

Yeah, you're right here: building from `main` would be terrible. We *could* detect the latest release commit, but that introduces lack of reproducibility, which is also my question about using the `slsa-verifier` installer action.

> Maybe add an install-latest: true type of flag input?

Looking back, I can't remember which camp I was in, but it makes more sense looking at it from this issue to have an option to use the latest release.

**Comment by laurentsimon** [comment]
I think we did not want to expose this option before because it breaks "pinning", ie we don't know which version would be installed by just reading the commit history. 

For example, if something breaks, users will create issues and it will be harder for us to figure which version of the verifier they used, since the run becomes non deterministic (we already have proper logging messages to identify the verifier, though). 

This would also have broken when we did not push the major version of the verifier, since the command would have changed. (although I think we can avoid this type of problem in the future, so not a strong argument).

So I see pros and cons for both. I feel like we could keep the code as-is for now since Sigstore is GA now and that the majority of breakage was due to the lack of Sigstore stability.

**Comment by ianlewis** [comment]
I agree we should focus on the core problem of sigstore stability and help them add tests older clients etc. rather than focusing on stop-gap measures in the generator workflows.
